src/cargo/util/hasher.rs

```diff
@@ -1,23 +1,24 @@
-//! Implementation of a hasher that produces the same values across releases.
+//! A hasher that produces the same values across releases and platforms.
 //!
-//! The hasher should be fast and have a low chance of collisions (but is not
-//! sufficient for cryptographic purposes).
-#![allow(deprecated)]
+//! This is a wrapper around [`rustc_stable_hash::StableHasher`].
 
-use std::hash::{Hasher, SipHasher};
-
-pub struct StableHasher(SipHasher);
+pub struct StableHasher(rustc_stable_hash::StableHasher);
 
 impl StableHasher {
     pub fn new() -> StableHasher {
-        StableHasher(SipHasher::new())
+        StableHasher(rustc_stable_hash::StableHasher::new())
+    }
+
+    pub fn finish(self) -> u64 {
+        self.0.finalize().0
     }
 }
 
-impl Hasher for StableHasher {
+impl std::hash::Hasher for StableHasher {
     fn finish(&self) -> u64 {
-        self.0.finish()
+        panic!("call StableHasher::finish instead");
     }
+
     fn write(&mut self, bytes: &[u8]) {
         self.0.write(bytes)
     }
```

Comment:
I'm not sure only forwarding `Hasher::write` is enough, since the endian-ness handling is done on the individual [`write_{u,i}{8,16,32,64,128}`](https://github.com/rust-lang/rustc-stable-hash/blob/cb8e141b08fb839606a5f79f9b56087cd54b764d/src/stable_hasher.rs#L81-L83) methods and not forwarding those will bypass that endian-ness handling.

I think it's also going to bypass the [`{u,i}size` handling](https://github.com/rust-lang/rustc-stable-hash/blob/cb8e141b08fb839606a5f79f9b56087cd54b764d/src/stable_hasher.rs#L135-L140).

Reply:
After rust-lang/rustc-stable-hash#6 and rust-lang/rustc-stable-hash#8, I think `StableSipHasher128` should be a drop-in replacement that we can use directly.

I am also thinking of blake3 as an alternative, but haven't figured out how to make it play nice with `ExtendedHasher`.